STM32Cube/Tasks/controller.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * controller.c
+ *
+ *  Created on: Apr 14, 2024
+ *      Author: joedo
+ */
+
+#include "controller.h"
+#include "freertos.h"
+#include "cmsis_os.h" //TODO replace with xtaskDelayUntil
+#include "stm32h7xx_hal.h"
+#include <stdio.h>
+#include <math.h>
+#include "millis.h"
+
+extern UART_HandleTypeDef huart4;
+
+
+//needed dependecy for stdio-->printf()
+int __io_putchar(int ch)
+{
+	HAL_UART_Transmit(&huart4,(uint8_t*)&ch,1,HAL_MAX_DELAY);
+	return ch;
+}
+
+
+
+void controlTask(void *argument)
+{
+	float altitudeEstimate;
+	controller_t airbrakesController;
+	TickType_t last_ticks = xTaskGetTickCount();
+
+	float startTime = millis();
+	osDelay(1000);
+	float measuredDelay = millis()-startTime;
+	printf("1000ms HAL_Delay is: %lu ms\r\n",(unsigned long)(round(measuredDelay)));
```

Comment:
comment out printfs for merging into main :sob:

Reply:
Just deleted all of this